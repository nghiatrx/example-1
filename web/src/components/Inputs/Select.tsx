import React from 'react';
import { Select as AntSelect } from 'antd';
const { Option } = AntSelect;

const Select = (props: SelectProps) => (
  <AntSelect value={props.input.value} onChange={(value: string) => props.input.onChange(value)}>
    {
      Array.isArray(props.options) && props.options.map(item => (
        <Option key={item.id} value={item.id}>{item.text}</Option>
      ))
    }
  </AntSelect>
)

export interface SelectProps {
  input: any;
  options: [{
    id: string;
    text: string;
  }]
}

export default Select;