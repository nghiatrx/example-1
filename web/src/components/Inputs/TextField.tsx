import React from 'react';
import { Input as AntInput } from 'antd';
import { useTranslation } from 'react-i18next';

const TextField = (props: TextField) => {
  const { input, meta: { error, warning } } = props;
  const { t } = useTranslation();

  return (
    <div>
      <AntInput
        value={input.value}
        onChange={e => input.onChange(e.target.value)} />
      {((error && <span className='error' style={{ color: '#ff0000' }}>{t(`${error}`)}</span>) || (warning && <span className='warning' color='#d4ff00'>{t(`${warning}`)}}</span>))}

    </div>
  )

}

export interface TextField extends HTMLInputElement {
  input: any,
  meta: { error: any, warning: any } // from reduxForm
}

export default TextField;