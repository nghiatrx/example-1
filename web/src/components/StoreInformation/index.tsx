import React from 'react';
import { useTranslation } from 'react-i18next';
import { Row, Col, Card } from 'antd';
import './style.scss';
import Information from './Information';

const StoreInformation = React.memo(() => {
  const { t } = useTranslation();
  return (
    <div>
      <h3 className='title'>{t('Store Information')}</h3>
      <Row gutter={[16, 16]}>
        <Col sm={24} md={10} lg={8}>
          <Information />
        </Col>

        <Col sm={24} md={14} lg={16}>
          <Card>
            <h5>{t('Delivery Default Message')}</h5>
          </Card>
        </Col>
      </Row>
    </div>
  );
})

export default StoreInformation;
