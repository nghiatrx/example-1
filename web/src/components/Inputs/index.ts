import TextField from './TextField';
import Select from './Select';

export { TextField };
export { Select };
