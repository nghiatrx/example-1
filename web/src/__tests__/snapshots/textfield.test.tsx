import React from 'react'
import { TextField } from '../../components/Inputs'
import renderer from 'react-test-renderer'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: key => key})
}))

afterEach(() => {
  jest.restoreAllMocks()
})

describe('Button', () => {
  it('should match to snapshot', () => {
    const component = renderer.create(<TextField input={{ value: 'test' }} meta={{ error: null, warning: null }} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})