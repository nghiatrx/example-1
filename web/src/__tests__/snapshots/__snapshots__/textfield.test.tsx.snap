// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match to snapshot 1`] = `
<div>
  <input
    className="ant-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="test"
  />
</div>
`;
